impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -352,49 +364,61 @@ public synchronized void startNodes(Predicate<Integer> nodeIdPredicate) {
     @SuppressWarnings("ResultOfMethodCallIgnored")
     @Override
     public synchronized void close() throws Exception {
-        try {
-            try {
-                roleOrderedShutdown(servers, false);
-            }
-            finally {
-                if (zooServer != null) {
-                    zooServer.shutdown(true);
-                }
-            }
-        }
-        finally {
-            if (tempDirectory.toFile().exists()) {
-                try (var ps = Files.walk(tempDirectory);
-                        var s = ps
-                                .sorted(Comparator.reverseOrder())
-                                .map(Path::toFile)) {
-                    s.forEach(File::delete);
-                }
-            }
-        }
+        roleOrderedShutdown(servers)
+                .whenComplete((unused, throwable) -> {
+                    if (throwable != null) {
+                        LOGGER.log(System.Logger.Level.ERROR, "error while shutting down kafka", throwable);
+                    }
+                    if (zooServer != null) {
+                        zooServer.shutdown(true);
+                    }
+                })
+                .whenComplete((unused, throwable) -> {
+                    if (throwable != null) {
+                        LOGGER.log(System.Logger.Level.ERROR, "error while shutting down zookeeper", throwable);
+                    }
+                    try {
+                        if (tempDirectory.toFile().exists()) {
+                            try (var ps = Files.walk(tempDirectory);
+                                    var s = ps
+                                            .sorted(Comparator.reverseOrder())
+                                            .map(Path::toFile)) {
+                                s.forEach(File::delete);
+                            }
+                        }
+                    }
+                    catch (IOException e) {
+                        LOGGER.log(System.Logger.Level.ERROR, "error while deleting temp files", throwable);
+                        throw new RuntimeException(e);
+                    }
+                });
```

Comment:
should we block on the cleanup? we could theoretically let it be cleaned in the background since each kafka/zk will use independent ports/tempdir. I think it would be cleaner to block.

Reply:
added in a blocking get